src/cryptography/sign.js

```diff
@@ -172,12 +172,19 @@ export const signAndPrintMessage = (message, passphrase, secondPassphrase) => {
 	return printSignedMessage(signedMessage);
 };
 
-export const signData = (data, passphrase) => {
+export const signDataWithPassphrase = (data, passphrase) => {
 	const { privateKey } = getPrivateAndPublicKeyBytesFromPassphrase(passphrase);
 	const signature = naclInstance.crypto_sign_detached(data, privateKey);
 	return bufferToHex(signature);
 };
 
+export const signDataWithPrivateKey = (data, privateKey) => {
```

Comment:
For symmetry reasons, I think `privateKey` should have the same type `publicKey` in `verifyData`, i.e. a hex string.

Reply:
@webmaster128  thanks 👍

Reply:
@webmaster128 in this case if `signDataWithPassphrase` use `signDataWithPrivateKey ` for data signing you need to convert `privateKey` to hex in `signDataWithPassphrase` function and convert it back to `Buffer` in the `signDataWithPrivateKey` because `crypto_sign_detached` expect Buffer in arguments not a hex. I think it a redundant logic.

Reply:
Buffer is probably the better type for signing as well as verifying. So the question is how much symmetry is valued over backward compatibility. Anyway I don't have a strong opinion here